### PR TITLE
LPS-66430

### PR DESCRIPTION
**Pull request by drewbrokke**
This is an update for [LPS-66430](https://issues.liferay.com/browse/LPS-66430).

@pei-jung can you sign off on this as well?

### CODE REVIEW

**Comment by JorgeFerrer** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: portal-web/docroot/html/taglib/ui/user_group_search_container_results/page.jsp [hunk 1]
```diff
@@ -25,44 +25,24 @@ SearchContainer userGroupSearchContainer = (SearchContainer)request.getAttribute
 if (Validator.isNotNull(searchTerms.getKeywords())) {
 	useIndexer = true;
 }
-
-Indexer<?> indexer = IndexerRegistryUtil.nullSafeGetIndexer(UserGroup.class);
 %>
 
 <liferay-ui:search-container id="<%= userGroupSearchContainer.getId(request, namespace) %>" searchContainer="<%= userGroupSearchContainer %>">
 	<liferay-ui:search-container-results>
-		<c:choose>
```
Comment (JorgeFerrer) [inline portal-web/docroot/html/taglib/ui/user_group_search_container_results/page.jsp]: Hey Drew, could you give me some more details on why this change is being done?

File: portal-impl/test/integration/com/liferay/portal/service/UserGroupLocalServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+	public static void tearDownClass() throws Exception {
+		RoleLocalServiceUtil.deleteRole(_role);
+		UserGroupLocalServiceUtil.deleteUserGroup(_userGroup1);
+		UserGroupLocalServiceUtil.deleteUserGroup(_userGroup2);
+	}
+
+	@Test
+	public void testSearchRoleUserGroups() {
+		List<UserGroup> userGroups = _doSearchUserGroups(
+			null, _userGroupParams);
```
Comment (JorgeFerrer) [inline portal-impl/test/integration/com/liferay/portal/service/UserGroupLocalServiceTest.java]: I think it would be better if each test defines its own params. Otherwise it's too hard to read. Also, you end up testing some things (such as the invalid key and value) in all tests.

Comment (JorgeFerrer) [inline portal-impl/test/integration/com/liferay/portal/service/UserGroupLocalServiceTest.java]: Also, passing "null" is not a very good practice since it reduces readability a lot. I would create to methods to do search one with keywords and one without and naming them accordingly.

File: portal-web/docroot/html/taglib/ui/user_group_search_container_results/page.jsp (hunk 1 above)
Comment (drewbrokke) [inline portal-web/docroot/html/taglib/ui/user_group_search_container_results/page.jsp]: The "useIndexer" attribute was a bit deceptive to begin with.  Even if a search was made with keywords, we still need to honor the params that are passed into the search function.  Doing an index search no matter what caused too many results to be returned.  In the case of this bug, we have keywords but do NOT want to use the index, so it made sense to use the search method, which would automatically determine when it was appropriate to use the indexer.

If we use `search` instead of `searchUserGroups`, the other two checks (`indexer.isIndexerEnabled() && PropsValues.USER_GROUPS_SEARCH_WITH_INDEX`) are automatically performed inside the service class in the search method, so they would be redundant here.

Comment (JorgeFerrer) [inline portal-web/docroot/html/taglib/ui/user_group_search_container_results/page.jsp]: Thanks for the explanation. It makes sense.

Other than the two comments I've made on tests, the pull looks good to me.
